SonarQube.Common/AnalysisConfig/AnalysisSetting.cs

```diff
@@ -27,5 +32,41 @@ public class AnalysisSetting
         /// </summary>
         [XmlAttribute]
         public string Value { get; set; }
+
+        #endregion
+
+        #region Static helper methods
+
+        /// <summary>
+        /// Regular expression to validate setting ids.
+        /// </summary>
+        /// <remarks>
+        /// Validation rules:
+        /// Must start with an alpanumeric character.
+        /// Can be followed by any number of alphanumeric characters or .
+        /// Whitespace is not allowed
+        /// </remarks>
+        private static readonly Regex ValidSettingKeyRegEx = new Regex(@"^\w[\w\d\.-]*$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Returns true if the supplied string is a valid key for a sonar-XXX.properties file, otherwise false
+        /// </summary>
+        public static bool IsValidKey(string key)
+        {
+            bool isValid = ValidSettingKeyRegEx.IsMatch(key);
+            return isValid;
+        }
+
+        /// <summary>
+        /// Comparer to use when comparing keys of analysis settings
+        /// </summary>
+        public static IEqualityComparer<string> SettingKeyComparer = StringComparer.Ordinal;
```

Comment:
Added a couple constants to give a standard way of comparing settings keys and values.
@dbolkensteyn: should the key/value settings be case sensitive or not?